### PR TITLE
TEST cache netmap github checkout

### PR DESCRIPTION
**Pull request by victorjulien**

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -2489,12 +2496,20 @@ jobs:
...
 
       - name: Checkout Netmap repository
+        if: steps.netmap-cache.outputs.cache-hit != 'true'
         uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
         with:
           repository: luigirizzo/netmap
           # gets cloned to $GITHUB_WORKSPACE/netmap/
           path: netmap/
 
+      - name: Save Netmap Cache
```
Comment (catenacyber) [inline .github/workflows/builds.yml:2506]: You could also cache the compilation

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -2489,12 +2496,20 @@ jobs:
                 linux-headers-$(uname -r)
 
       - name: Checkout Netmap repository
+        if: steps.netmap-cache.outputs.cache-hit != 'true'
         uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
```
Comment (catenacyber) [inline .github/workflows/builds.yml:2500]: `git clone` seems to work better than `actions/checkout` in https://github.com/OISF/suricata/pull/11903

But not sure it is a good path to follow

File: .github/workflows/builds.yml (hunk 1 above)
Comment (victorjulien) [reply .github/workflows/builds.yml:2506]: I don't want to add additional complexity, this is just about avoiding the rate limit issue

File: .github/workflows/builds.yml (hunk 2 above)
Comment (victorjulien) [reply .github/workflows/builds.yml:2500]: I think the checkout action is more idiomatic

**Comment by victorjulien** [comment]
replaced by #11918

**Pull request closed without merging**